pkg/virt-controller/watch/vm.go

```diff
@@ -1419,31 +1419,52 @@ func (c *VMController) isVirtualMachineStatusMigrating(vm *virtv1.VirtualMachine
 }
 
 func (c *VMController) syncReadyConditionFromVMI(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachineInstance) {
-	vmReadyCond := controller.NewVirtualMachineConditionManager().
-		GetCondition(vm, virtv1.VirtualMachineReady)
-	vmiReadyCond := controller.NewVirtualMachineInstanceConditionManager().
-		GetCondition(vmi, virtv1.VirtualMachineInstanceConditionType(k8score.PodReady))
-
-	if vmReadyCond == nil && vmiReadyCond != nil {
-		log.Log.Object(vm).V(4).Info("Adding ready condition")
-		newCond := virtv1.VirtualMachineCondition{Type: virtv1.VirtualMachineReady}
-		copyConditionDetails(vmiReadyCond, &newCond)
-		vm.Status.Conditions = append(vm.Status.Conditions, newCond)
-	} else if vmReadyCond != nil && vmiReadyCond != nil {
-		log.Log.Object(vm).V(4).Info("Updating ready condition")
-		copyConditionDetails(vmiReadyCond, vmReadyCond)
-	} else if vmReadyCond != nil && vmiReadyCond == nil {
-		log.Log.Object(vm).V(4).Info("Removing ready condition")
+	setVMReadyCondition := func(status k8score.ConditionStatus, reason, message string, lastProbeTime, lastTransitionTime v1.Time) {
```

Comment:
could we move this in a dedicated function to make this whole method more readable?